{"name": "部门里添加一人作为无法删除的条件", "status": "failed", "statusDetails": {"message": "AssertionError: 断言失败 -|- 实际结果:用户帐号已存在 || 预期结果: 成功", "trace": "cls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '220001', 'errmsg': '用户帐号已存在'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n>               assert actual == v\nE               AssertionError\n\ntools/data_process.py:118: AssertionError\n\nDuring handling of the above exception, another exception occurred:\n\ncases = ['case_012', '测试数据准备', '成员，部门相关', '部门里添加一人作为无法删除的条件', '/v1/user/create?access_token=&$.case_001.access_token&', '', ...]\n\n    @pytest.mark.flaky(reruns=0, reruns_delay=1)\n    def test_main(cases):#cases, get_db\n        # 此处的cases入参来自与 conftest.py  文件中 cases函数，与直接使用 @pytest.mark.parametrize\n        # 有着差不多的效果\n        # 发送请求\n        response, expect, sql = BaseRequest.send_request(cases)\n        # 执行sql\n        # DataProcess.handle_sql(sql, get_db)\n        # 断言操作\n>       DataProcess.assert_result(response, expect)\n\ntest/test_api.py:29: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\ncls = <class 'tools.data_process.DataProcess'>\nresponse = {'errcode': '220001', 'errmsg': '用户帐号已存在'}\nexpect_str = '{\"$.errmsg\":\"成功\"}'\n\n    @classmethod\n    def assert_result(cls, response: dict, expect_str: str):\n        \"\"\" 预期结果实际结果断言方法\n        :param response: 实际响应结果\n        :param expect_str: 预期响应内容，从excel中读取\n        return None\n        \"\"\"\n        # 后置sql变量转换\n        expect_str = rep_expr(expect_str, DataProcess.response_dict)\n        expect_dict = convert_json(expect_str)\n        index = 0\n        for k, v in expect_dict.items():\n            # 获取需要断言的实际结果部分\n            actual = extractor(response, k)\n            index += 1\n            logger.info(f'第{index}个断言,实际结果:{actual} | 预期结果:{v} \\n断言结果 {actual == v}')\n            allure_step(f'第{index}个断言',  f'实际结果:{actual} = 预期结果:{v}')\n            try:\n                assert actual == v\n            except AssertionError:\n>               raise AssertionError(f'断言失败 -|- 实际结果:{actual} || 预期结果: {v}')\nE               AssertionError: 断言失败 -|- 实际结果:用户帐号已存在 || 预期结果: 成功\n\ntools/data_process.py:120: AssertionError"}, "steps": [{"name": "请求地址", "status": "passed", "attachments": [{"name": "请求地址", "source": "ae66b45b-fe6f-4e41-8097-6b25fcac88bc-attachment.txt", "type": "text/plain"}], "start": 1622542318784, "stop": 1622542318785}, {"name": "请求头", "status": "passed", "attachments": [{"name": "请求头", "source": "f50cf473-b4d8-4657-bd73-ffcebc7ba1b4-attachment.txt", "type": "text/plain"}], "start": 1622542318785, "stop": 1622542318785}, {"name": "请求参数", "status": "passed", "attachments": [{"name": "请求参数", "source": "b3db5819-6163-466b-b568-f0f626dabb08-attachment.txt", "type": "text/plain"}], "start": 1622542318785, "stop": 1622542318785}, {"name": "上传文件", "status": "passed", "attachments": [{"name": "上传文件", "source": "0abb4cae-0afb-4b0f-b6e1-3e89c9c81c0b-attachment.txt", "type": "text/plain"}], "start": 1622542318785, "stop": 1622542318785}, {"name": "响应耗时(s)", "status": "passed", "attachments": [{"name": "响应耗时(s)", "source": "2a82f247-bd51-4a17-973d-0e324d0f13b1-attachment.txt", "type": "text/plain"}], "start": 1622542319084, "stop": 1622542319084}, {"name": "响应内容", "status": "passed", "attachments": [{"name": "响应内容", "source": "2804818b-48cc-4535-8373-44665e6a12a6-attachment.txt", "type": "text/plain"}], "start": 1622542319084, "stop": 1622542319084}, {"name": "存储实际响应", "status": "passed", "attachments": [{"name": "存储实际响应", "source": "bd0605d4-1d5f-4d13-a25b-7eaae3d074f0-attachment.txt", "type": "text/plain"}], "start": 1622542319085, "stop": 1622542319085}, {"name": "第1个断言", "status": "passed", "attachments": [{"name": "第1个断言", "source": "68d0eb8b-41ac-40e0-aef0-7442ac6587a4-attachment.txt", "type": "text/plain"}], "start": 1622542319085, "stop": 1622542319085}], "parameters": [{"name": "cases", "value": "['case_012', '测试数据准备', '成员，部门相关', '部门里添加一人作为无法删除的条件', '/v1/user/create?access_token=&$.case_001.access_token&', '', 'post', 'json', '', '{\\n    \"name\": \"部门里添加一人\",\\n    \"department\": [\\n        {\\n            \"did\": \"&$.case_009.did&\",\\n            \"order\": \"2\",\\n            \"posName\": \"ceo\"\\n        }\\n    ],\\n    \"gender\": \"1\",\\n    \"sign\": \"故意添加一位作为群无法删除的条件\",\\n    \"enName\": \"lucky\",\\n    \"tel\": \"0215553222\",\\n    \"mobile\": \"15000000000\"\\n}', '', '', '{\"$.errmsg\":\"成功\"}']"}], "start": 1622542318784, "stop": 1622542319085, "uuid": "9c2f6d40-5ac4-4d81-8d83-bbef5b940684", "historyId": "5a1a59aa4470006c7c58a5ba625a4478", "testCaseId": "881516e7ef2e2aa5436f6a3c3e6a5bcb", "fullName": "test.test_api#test_main", "labels": [{"name": "tag", "value": "flaky(reruns=0, reruns_delay=1)"}, {"name": "parentSuite", "value": "test"}, {"name": "suite", "value": "test_api"}, {"name": "host", "value": "liuchengxudeMacBook-Pro.local"}, {"name": "thread", "value": "19614-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "test.test_api"}]}